### ISSUE TITLE
Multi-hop tunneling?

### ISSUE THREAD

**Issue by staltz** [ssbc/ssb-tunnel#2]
Since this plugin is the same for both portals and clients, it seems like clients could also become portals, so there is the prospect of doing:

```
,---,      ,---,     ,---,     ,---,
|   |----->|   |<----|   |<----|   |
| A |<=====|-B-|<====| C |<====| D |
|   |----->|   |<----|   |<----|   |
`---`      `---`     `---`     `---`
```

And effectively having a tunnel D-(C)-(B)->A. Is this a real possibility? Is this a good idea or a bad idea? I can imagine it being possible in Bluetooth, at least.

I don't intend to build this, but I stumbled upon this theoretical possibility and it made me wonder that there is no way I can *prevent* it from happening. Or is there?

**Comment by dominictarr** [comment]
yes, this is possible. no I don't think there is any way you can prevent it. and also, it would be useful for somethings!

hmm, in the first version of this, you supplied a whole address to the portal, but then I refactored it so that you just said the id you want to connect to as the portal (and it requires you already have an address for them in the gossip table) that means you could connect to a portal that your address for them is a tunnel address, so this might already work?

**Comment by staltz** [comment]
I really hate this bot

**Comment by staltz** [comment]
I hate you.

**Comment by christianbundy** [comment]
@staltz 

If you assign yourself to an issue, you can vouch for it to stay open and stale[bot] won't even think about bothering you. If you later decide you don't care about it, you can unassign and stale[bot] will treat it like a normal issue. It's like a subjective roadmap, where the assignees are the people who care about an issue.
